charms/kfp-api/src/charm.py

```diff
@@ -142,11 +139,38 @@ def __init__(self, *args):
             ],
         )
 
+    @property
+    def _charm_default_kubernetes_labels(self):
+        """Return default labels for Kubernetes resources."""
+        return create_charm_default_labels(
+            application_name=self.model.app.name,
+            model_name=self.model.name,
+            scope="all-resources",
+        )
+
     @property
     def container(self):
         """Return container."""
         return self._container
 
+    @property
+    def _context(self):
+        """Return the context used for generating kubernetes resources."""
+        interfaces = self._get_interfaces()
+        os = self._get_object_storage(interfaces)
+
+        context = {
+            "app_name": self._name,
+            "namespace": self._namespace,
+            "service": self._name,
+            "grpc_port": self._grcp_port,
+            "http_port": self._http_port,
+            # Must include .svc.cluster.local for DNS resolution
+            "minio_url": f"{os['service']}.{os['namespace']}.svc.cluster.local",
```

Comment:
The hardcoded value in upstream expects the Namespace value to always be `kubeflow` for the minio Service. I wonder if we should add a check in the kfp-api charm code to ensure the object store shares this expected value, otherwise we can block the unit. I wouldn't expect this to happen often as these two charms should be deployed in the same model, but I wonder if we should ensure this on the "requirer" side.

Reply:
Good point.  What do you think about adding [this check](https://github.com/canonical/kubeflow-dashboard-operator/blob/a6f9dc0968ded793a1692e10766e4454f4c9127d/src/charm.py#L184) to kfp-api, and then hard coding this to `kubeflow` and adding a note?

Reply:
Added the check - let me know what you think